{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This script produces a table of PMC paper IDs matched to the SRA or GEO datasets they contain, and metadata about those datasets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import required modules\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from collections import Counter\n",
    "import statistics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "STEP I: Clean the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Series</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Datasets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>GPL4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>GSM2</td>\n",
       "      <td>GPL4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>GSE462</td>\n",
       "      <td>GSM3</td>\n",
       "      <td>GPL5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>GSE462</td>\n",
       "      <td>GSM4</td>\n",
       "      <td>GPL5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>GSE462</td>\n",
       "      <td>GSM5</td>\n",
       "      <td>GPL5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3280011</td>\n",
       "      <td>GSE136775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3280012</td>\n",
       "      <td>GSE136776</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3280013</td>\n",
       "      <td>GSE137458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3280014</td>\n",
       "      <td>GSE137562</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3280015</td>\n",
       "      <td>GSE137563</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3280016 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Series Accession Platform Datasets\n",
       "0           GSE506      GSM1     GPL4      NaN\n",
       "1           GSE506      GSM2     GPL4      NaN\n",
       "2           GSE462      GSM3     GPL5      NaN\n",
       "3           GSE462      GSM4     GPL5      NaN\n",
       "4           GSE462      GSM5     GPL5      NaN\n",
       "...            ...       ...      ...      ...\n",
       "3280011  GSE136775       NaN      NaN      NaN\n",
       "3280012  GSE136776       NaN      NaN      NaN\n",
       "3280013  GSE137458       NaN      NaN      NaN\n",
       "3280014  GSE137562       NaN      NaN      NaN\n",
       "3280015  GSE137563       NaN      NaN      NaN\n",
       "\n",
       "[3280016 rows x 4 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# import GEO reference data\n",
    "\n",
    "# import table with accession, platform, and series\n",
    "geoAPS = pd.read_csv('../geo_sample.csv')\n",
    "\n",
    "# import table with datasets IDs\n",
    "geoDS = pd.read_csv('../geo_series_table.csv', low_memory = False)\n",
    "geoDS.columns = ['Series', 'Title', 'Series Type', 'Taxonomy', \n",
    "                 'Sample Count', 'Datasets', 'Supplementary Types', \n",
    "                 'Supplementary Links', 'PubMed ID', 'SRA Accession', \n",
    "                 'Contact', 'Release Date']\n",
    "\n",
    "# add datasets column by merging\n",
    "allData = pd.merge(geoAPS, geoDS, how = 'outer', on = 'Series')\n",
    "geoReference = allData[['Series', 'Accession', 'Platform', 'Datasets']].drop_duplicates()\n",
    "\n",
    "geoReference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SRAStudy</th>\n",
       "      <th>Run</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>BioProject</th>\n",
       "      <th>Submission</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ERP000767</td>\n",
       "      <td>ERR3550121</td>\n",
       "      <td>ERX3556726</td>\n",
       "      <td>PRJEB2600</td>\n",
       "      <td>ERA2143351</td>\n",
       "      <td>ERS3773255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ERP000767</td>\n",
       "      <td>ERR3550142</td>\n",
       "      <td>ERX3556747</td>\n",
       "      <td>PRJEB2600</td>\n",
       "      <td>ERA2143351</td>\n",
       "      <td>ERS3773255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ERP000767</td>\n",
       "      <td>ERR3550147</td>\n",
       "      <td>ERX3556752</td>\n",
       "      <td>PRJEB2600</td>\n",
       "      <td>ERA2143351</td>\n",
       "      <td>ERS3773263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ERP000767</td>\n",
       "      <td>ERR3550125</td>\n",
       "      <td>ERX3556730</td>\n",
       "      <td>PRJEB2600</td>\n",
       "      <td>ERA2143351</td>\n",
       "      <td>ERS3773263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>ERP000767</td>\n",
       "      <td>ERR3550137</td>\n",
       "      <td>ERX3556742</td>\n",
       "      <td>PRJEB2600</td>\n",
       "      <td>ERA2143351</td>\n",
       "      <td>ERS3773252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7676058</td>\n",
       "      <td>SRP002493</td>\n",
       "      <td>SRR049674</td>\n",
       "      <td>SRX020916</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRA017695</td>\n",
       "      <td>SRS073299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7676059</td>\n",
       "      <td>SRP002493</td>\n",
       "      <td>SRR049675</td>\n",
       "      <td>SRX021028</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRA017695</td>\n",
       "      <td>SRS073299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7676060</td>\n",
       "      <td>SRP002493</td>\n",
       "      <td>SRR049677</td>\n",
       "      <td>SRX021029</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRA017695</td>\n",
       "      <td>SRS073299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7676061</td>\n",
       "      <td>SRP002064</td>\n",
       "      <td>SRR042496</td>\n",
       "      <td>SRX020084</td>\n",
       "      <td>PRJNA46359</td>\n",
       "      <td>DIEGO</td>\n",
       "      <td>SRS011854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7676062</td>\n",
       "      <td>SRP002438</td>\n",
       "      <td>SRR043092</td>\n",
       "      <td>SRX020242</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OF</td>\n",
       "      <td>SRS072567</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7662363 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          SRAStudy         Run  Experiment  BioProject  Submission      Sample\n",
       "0        ERP000767  ERR3550121  ERX3556726   PRJEB2600  ERA2143351  ERS3773255\n",
       "1        ERP000767  ERR3550142  ERX3556747   PRJEB2600  ERA2143351  ERS3773255\n",
       "2        ERP000767  ERR3550147  ERX3556752   PRJEB2600  ERA2143351  ERS3773263\n",
       "3        ERP000767  ERR3550125  ERX3556730   PRJEB2600  ERA2143351  ERS3773263\n",
       "4        ERP000767  ERR3550137  ERX3556742   PRJEB2600  ERA2143351  ERS3773252\n",
       "...            ...         ...         ...         ...         ...         ...\n",
       "7676058  SRP002493   SRR049674   SRX020916         NaN   SRA017695   SRS073299\n",
       "7676059  SRP002493   SRR049675   SRX021028         NaN   SRA017695   SRS073299\n",
       "7676060  SRP002493   SRR049677   SRX021029         NaN   SRA017695   SRS073299\n",
       "7676061  SRP002064   SRR042496   SRX020084  PRJNA46359       DIEGO   SRS011854\n",
       "7676062  SRP002438   SRR043092   SRX020242         NaN          OF   SRS072567\n",
       "\n",
       "[7662363 rows x 6 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# import SRA reference data\n",
    "sraReference = pd.read_csv('../sraIDfull.csv', error_bad_lines = False, low_memory=False, quoting=3)\n",
    "sraReference = sraReference[['SRAStudy', 'Run', 'Experiment', \n",
    "                             'BioProject', 'Submission', 'Sample']].drop_duplicates()\n",
    "sraReference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>pmc_ID</th>\n",
       "      <th>accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3707052</td>\n",
       "      <td>GSM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3706879</td>\n",
       "      <td>GSE45534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4255636</td>\n",
       "      <td>GSE5281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE67036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE1297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>144400</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589301</td>\n",
       "      <td>GSE62254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>144401</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>144403</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>144404</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>PRJNA437812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>144405</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>SRP135146</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78620 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    journal      pmc_ID    accession\n",
       "0       Alzheimers_Res_Ther  PMC3707052         GSM1\n",
       "1       Alzheimers_Res_Ther  PMC3706879     GSE45534\n",
       "5       Alzheimers_Res_Ther  PMC4255636      GSE5281\n",
       "6       Alzheimers_Res_Ther  PMC4731966     GSE67036\n",
       "7       Alzheimers_Res_Ther  PMC4731966      GSE1297\n",
       "...                     ...         ...          ...\n",
       "144400          Dis_Markers  PMC6589301     GSE62254\n",
       "144401          Dis_Markers  PMC3834650      GDS1059\n",
       "144403          Dis_Markers  PMC3834650         GDS1\n",
       "144404     Case_Rep_Hematol  PMC6462343  PRJNA437812\n",
       "144405     Case_Rep_Hematol  PMC6462343    SRP135146\n",
       "\n",
       "[78620 rows x 3 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# import data scraped from PubMed\n",
    "# this data was scraped from XML files on the hoffman2 cluster: /u/scratch/n/nikodm/pmcOA/\n",
    "\n",
    "pmcData = pd.read_csv('../data_tables/preFilterMatrix.csv').drop_duplicates()\n",
    "pmcData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>pmc_ID</th>\n",
       "      <th>accession</th>\n",
       "      <th>geoSeries</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3707052</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>GSE506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3706879</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GSE45534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4255636</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GSE5281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GSE67036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GSE1297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78615</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589301</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GSE62254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78616</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1059</td>\n",
       "      <td>GSE2191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78617</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78618</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>PRJNA437812</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78619</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78620 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   journal      pmc_ID    accession geoSeries\n",
       "0      Alzheimers_Res_Ther  PMC3707052         GSM1    GSE506\n",
       "1      Alzheimers_Res_Ther  PMC3706879     GSE45534  GSE45534\n",
       "2      Alzheimers_Res_Ther  PMC4255636      GSE5281   GSE5281\n",
       "3      Alzheimers_Res_Ther  PMC4731966     GSE67036  GSE67036\n",
       "4      Alzheimers_Res_Ther  PMC4731966      GSE1297   GSE1297\n",
       "...                    ...         ...          ...       ...\n",
       "78615          Dis_Markers  PMC6589301     GSE62254  GSE62254\n",
       "78616          Dis_Markers  PMC3834650      GDS1059   GSE2191\n",
       "78617          Dis_Markers  PMC3834650         GDS1       NaN\n",
       "78618     Case_Rep_Hematol  PMC6462343  PRJNA437812       NaN\n",
       "78619     Case_Rep_Hematol  PMC6462343    SRP135146       NaN\n",
       "\n",
       "[78620 rows x 4 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge GEO accessions with reference data, convert to Series\n",
    "\n",
    "# match series first\n",
    "se = pd.merge(pmcData, geoReference['Series'].drop_duplicates(), how = 'left',\n",
    "            left_on = 'accession', right_on = 'Series')\n",
    "pmcData = se.rename(columns = {'Series': 'Series_result'})\n",
    "\n",
    "# match each other style of GEO ID\n",
    "for col in ['Accession', 'Platform', 'Datasets']:\n",
    "    pmcData = pd.merge(pmcData, geoReference[['Series', col]].drop_duplicates(subset = col), how = 'left', \n",
    "            left_on = 'accession', right_on = col)\n",
    "    label = col + '_result'\n",
    "    pmcData = pmcData.rename(columns = {'Series': label})\n",
    "\n",
    "# combine all GEO series match columns into one aggregate GEO series column, clean up\n",
    "pmcData['geoSeries'] = pmcData['Series_result'].fillna(pmcData['Accession_result']).fillna(pmcData['Platform_result']).fillna(pmcData['Datasets_result'])\n",
    "pmcData = pmcData.drop(labels = ['Accession', 'Platform', 'Datasets',\n",
    "                           'Series_result', 'Accession_result',\n",
    "                           'Platform_result', 'Datasets_result'], axis = 1)\n",
    "pmcData_geoMerged = pmcData\n",
    "pmcData_geoMerged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>pmc_ID</th>\n",
       "      <th>accession</th>\n",
       "      <th>geoSeries</th>\n",
       "      <th>sraStudy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3707052</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3706879</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4255636</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78615</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589301</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78616</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1059</td>\n",
       "      <td>GSE2191</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78617</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78618</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>PRJNA437812</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRP135146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78619</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRP135146</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78620 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   journal      pmc_ID    accession geoSeries   sraStudy\n",
       "0      Alzheimers_Res_Ther  PMC3707052         GSM1    GSE506        NaN\n",
       "1      Alzheimers_Res_Ther  PMC3706879     GSE45534  GSE45534        NaN\n",
       "2      Alzheimers_Res_Ther  PMC4255636      GSE5281   GSE5281        NaN\n",
       "3      Alzheimers_Res_Ther  PMC4731966     GSE67036  GSE67036        NaN\n",
       "4      Alzheimers_Res_Ther  PMC4731966      GSE1297   GSE1297        NaN\n",
       "...                    ...         ...          ...       ...        ...\n",
       "78615          Dis_Markers  PMC6589301     GSE62254  GSE62254        NaN\n",
       "78616          Dis_Markers  PMC3834650      GDS1059   GSE2191        NaN\n",
       "78617          Dis_Markers  PMC3834650         GDS1       NaN        NaN\n",
       "78618     Case_Rep_Hematol  PMC6462343  PRJNA437812       NaN  SRP135146\n",
       "78619     Case_Rep_Hematol  PMC6462343    SRP135146       NaN  SRP135146\n",
       "\n",
       "[78620 rows x 5 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge SRA accessions with reference data, convert to Study\n",
    "\n",
    "# match SRA Study IDs first\n",
    "st = pd.merge(pmcData_geoMerged, sraReference['SRAStudy'].drop_duplicates(), how = 'left',\n",
    "            left_on = 'accession', right_on = 'SRAStudy')\n",
    "pmcData = st.rename(columns = {'SRAStudy': 'Study_result'})\n",
    "\n",
    "# match every other style of SRA ID\n",
    "for col in ['Run', 'Experiment', 'BioProject', 'Submission', 'Sample']:\n",
    "    pmcData = pd.merge(pmcData, sraReference[['SRAStudy', col]].drop_duplicates(subset = [col]), how = 'left',\n",
    "                      left_on = 'accession', right_on = col)\n",
    "    label = col + '_result'\n",
    "    pmcData = pmcData.rename(columns = {'SRAStudy': label})\n",
    "\n",
    "# combine all SRA Study matches into one aggregate column, clean up\n",
    "pmcData['sraStudy'] = pmcData['Study_result'].fillna(pmcData['Run_result']).fillna(pmcData['Experiment_result']).fillna(pmcData['BioProject_result']).fillna(pmcData['Submission_result']).fillna(pmcData['Sample_result'])\n",
    "pmcData = pmcData.drop(labels = ['Run', 'Experiment', 'BioProject', 'Submission', 'Sample',\n",
    "                                'Study_result', 'Run_result', 'Experiment_result', 'BioProject_result',\n",
    "                                'Submission_result', 'Sample_result'], axis = 1)\n",
    "\n",
    "pmcData_sraMerged = pmcData\n",
    "pmcData_sraMerged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>pmc_ID</th>\n",
       "      <th>accession</th>\n",
       "      <th>converted_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3707052</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>GSE506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3706879</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GSE45534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4255636</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GSE5281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GSE67036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GSE1297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78615</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589301</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GSE62254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78616</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1059</td>\n",
       "      <td>GSE2191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78617</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78618</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>PRJNA437812</td>\n",
       "      <td>SRP135146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78619</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRP135146</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78620 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   journal      pmc_ID    accession converted_accession\n",
       "0      Alzheimers_Res_Ther  PMC3707052         GSM1              GSE506\n",
       "1      Alzheimers_Res_Ther  PMC3706879     GSE45534            GSE45534\n",
       "2      Alzheimers_Res_Ther  PMC4255636      GSE5281             GSE5281\n",
       "3      Alzheimers_Res_Ther  PMC4731966     GSE67036            GSE67036\n",
       "4      Alzheimers_Res_Ther  PMC4731966      GSE1297             GSE1297\n",
       "...                    ...         ...          ...                 ...\n",
       "78615          Dis_Markers  PMC6589301     GSE62254            GSE62254\n",
       "78616          Dis_Markers  PMC3834650      GDS1059             GSE2191\n",
       "78617          Dis_Markers  PMC3834650         GDS1                 NaN\n",
       "78618     Case_Rep_Hematol  PMC6462343  PRJNA437812           SRP135146\n",
       "78619     Case_Rep_Hematol  PMC6462343    SRP135146           SRP135146\n",
       "\n",
       "[78620 rows x 4 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# combine GEO Series hits and SRA study hits into one converted accession column\n",
    "pmcData['converted_accession'] = pmcData_sraMerged['geoSeries'].fillna(pmcData_sraMerged['sraStudy'])\n",
    "pmcData = pmcData.drop(labels = ['geoSeries', 'sraStudy'], axis = 1)\n",
    "pmcData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>pmc_ID</th>\n",
       "      <th>accession</th>\n",
       "      <th>converted_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3707052</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>GSE506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3706879</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GSE45534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4255636</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GSE5281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GSE67036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GSE1297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78614</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589307</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GSE14795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78615</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589301</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GSE62254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78616</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1059</td>\n",
       "      <td>GSE2191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78618</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>PRJNA437812</td>\n",
       "      <td>SRP135146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78619</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRP135146</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73490 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   journal      pmc_ID    accession converted_accession\n",
       "0      Alzheimers_Res_Ther  PMC3707052         GSM1              GSE506\n",
       "1      Alzheimers_Res_Ther  PMC3706879     GSE45534            GSE45534\n",
       "2      Alzheimers_Res_Ther  PMC4255636      GSE5281             GSE5281\n",
       "3      Alzheimers_Res_Ther  PMC4731966     GSE67036            GSE67036\n",
       "4      Alzheimers_Res_Ther  PMC4731966      GSE1297             GSE1297\n",
       "...                    ...         ...          ...                 ...\n",
       "78614          Dis_Markers  PMC6589307     GSE14795            GSE14795\n",
       "78615          Dis_Markers  PMC6589301     GSE62254            GSE62254\n",
       "78616          Dis_Markers  PMC3834650      GDS1059             GSE2191\n",
       "78618     Case_Rep_Hematol  PMC6462343  PRJNA437812           SRP135146\n",
       "78619     Case_Rep_Hematol  PMC6462343    SRP135146           SRP135146\n",
       "\n",
       "[73490 rows x 4 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# clean out erroneous converted_accession entries, and rows that didn't map to a converted_accession\n",
    "w = []\n",
    "for a in pmcData['converted_accession']:\n",
    "    if(type(a) == str):\n",
    "        if a[0:3] != 'GSE' and a[0:3] != 'GPL' and a[1:3] != 'RP' and a not in w:\n",
    "                w.append(a)\n",
    "w.append(np.NaN)\n",
    "        \n",
    "pmcData = pmcData[~pmcData.converted_accession.isin(w)]\n",
    "pmcData"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Step II: Collect all desired factors about each SRA or GEO dataset, and add them to the table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define functions to convert dates to a universal format\n",
    "def strToMonth(m):\n",
    "    if(m == 'Jan'):\n",
    "        return '01'\n",
    "    elif(m == 'Feb'):\n",
    "        return '02'\n",
    "    elif(m == 'Mar'):\n",
    "        return '03'\n",
    "    elif(m == 'Apr'):\n",
    "        return '04'\n",
    "    elif(m == 'May'):\n",
    "        return '05'\n",
    "    elif(m == 'Jun'):\n",
    "        return '06'\n",
    "    elif(m == 'Jul'):\n",
    "        return '07'\n",
    "    elif(m == 'Aug'):\n",
    "        return '08'\n",
    "    elif(m == 'Sep'):\n",
    "        return '09'\n",
    "    elif(m == 'Oct'):\n",
    "        return '10'\n",
    "    elif(m == 'Nov'):\n",
    "        return '11'\n",
    "    elif(m == 'Dec'):\n",
    "        return '12'\n",
    "    else:\n",
    "        return(np.NaN)\n",
    "    \n",
    "def convGEODate(d):\n",
    "    if(type(d) == str):\n",
    "        mon = strToMonth(d[0:3])\n",
    "        day = d[4:6]\n",
    "        yr = d[8:12]\n",
    "        return yr + '-' + mon + '-' + day\n",
    "    else:\n",
    "        return np.NaN\n",
    "\n",
    "# def convSRADate(d):\n",
    "#     return d[0:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import SRA attribute data\n",
    "# CAUTION: large file! Time delay on import...\n",
    "\n",
    "sraAttributes = pd.read_csv('../sra_complete_runs.csv', error_bad_lines = False, low_memory=False)    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>ReleaseDate</th>\n",
       "      <th>LoadDate</th>\n",
       "      <th>spots</th>\n",
       "      <th>bases</th>\n",
       "      <th>spots_with_mates</th>\n",
       "      <th>avgLength</th>\n",
       "      <th>size_MB</th>\n",
       "      <th>AssemblyName</th>\n",
       "      <th>download_path</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>LibraryName</th>\n",
       "      <th>LibraryStrategy</th>\n",
       "      <th>LibrarySelection</th>\n",
       "      <th>LibrarySource</th>\n",
       "      <th>LibraryLayout</th>\n",
       "      <th>InsertSize</th>\n",
       "      <th>InsertDev</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Model</th>\n",
       "      <th>SRAStudy</th>\n",
       "      <th>BioProject</th>\n",
       "      <th>Study_Pubmed_id</th>\n",
       "      <th>ProjectID</th>\n",
       "      <th>Sample</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>SampleType</th>\n",
       "      <th>TaxID</th>\n",
       "      <th>ScientificName</th>\n",
       "      <th>SampleName</th>\n",
       "      <th>g1k_pop_code</th>\n",
       "      <th>source</th>\n",
       "      <th>g1k_analysis_group</th>\n",
       "      <th>Subject_ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Disease</th>\n",
       "      <th>Tumor</th>\n",
       "      <th>Affection_Status</th>\n",
       "      <th>Analyte_Type</th>\n",
       "      <th>Histological_Type</th>\n",
       "      <th>Body_Site</th>\n",
       "      <th>CenterName</th>\n",
       "      <th>Submission</th>\n",
       "      <th>dbgap_study_accession</th>\n",
       "      <th>Consent</th>\n",
       "      <th>RunHash</th>\n",
       "      <th>ReadHash</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ERR3550121</td>\n",
       "      <td>2019-09-25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERX3556726</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>ERP000767</td>\n",
       "      <td>PRJEB2600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204335</td>\n",
       "      <td>ERS3773255</td>\n",
       "      <td>NaN</td>\n",
       "      <td>simple</td>\n",
       "      <td>0</td>\n",
       "      <td>Leishmania infantum</td>\n",
       "      <td>0c4ac31a-dec5-11e9-ba7b-68b599768938</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UOC-CCG</td>\n",
       "      <td>ERA2143351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ERR3550142</td>\n",
       "      <td>2019-09-25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERX3556747</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>ERP000767</td>\n",
       "      <td>PRJEB2600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204335</td>\n",
       "      <td>ERS3773255</td>\n",
       "      <td>NaN</td>\n",
       "      <td>simple</td>\n",
       "      <td>0</td>\n",
       "      <td>Leishmania infantum</td>\n",
       "      <td>0c4ac31a-dec5-11e9-ba7b-68b599768938</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UOC-CCG</td>\n",
       "      <td>ERA2143351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ERR3550147</td>\n",
       "      <td>2019-09-25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERX3556752</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>ERP000767</td>\n",
       "      <td>PRJEB2600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204335</td>\n",
       "      <td>ERS3773263</td>\n",
       "      <td>NaN</td>\n",
       "      <td>simple</td>\n",
       "      <td>0</td>\n",
       "      <td>Leishmania donovani</td>\n",
       "      <td>0e2d3a2c-dec6-11e9-a5d1-68b59976a384</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UOC-CCG</td>\n",
       "      <td>ERA2143351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ERR3550125</td>\n",
       "      <td>2019-09-25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERX3556730</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>ERP000767</td>\n",
       "      <td>PRJEB2600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204335</td>\n",
       "      <td>ERS3773263</td>\n",
       "      <td>NaN</td>\n",
       "      <td>simple</td>\n",
       "      <td>0</td>\n",
       "      <td>Leishmania donovani</td>\n",
       "      <td>0e2d3a2c-dec6-11e9-a5d1-68b59976a384</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UOC-CCG</td>\n",
       "      <td>ERA2143351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>ERR3550137</td>\n",
       "      <td>2019-09-25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERX3556742</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>ERP000767</td>\n",
       "      <td>PRJEB2600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204335</td>\n",
       "      <td>ERS3773252</td>\n",
       "      <td>NaN</td>\n",
       "      <td>simple</td>\n",
       "      <td>0</td>\n",
       "      <td>Leishmania donovani</td>\n",
       "      <td>188c9a6e-deba-11e9-ba7b-68b599768938</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UOC-CCG</td>\n",
       "      <td>ERA2143351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7676058</td>\n",
       "      <td>SRR049674</td>\n",
       "      <td>2010-05-27</td>\n",
       "      <td>2014-05-27 02:57:13</td>\n",
       "      <td>35888370</td>\n",
       "      <td>1794418500</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>1504</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>SRX020916</td>\n",
       "      <td>VVWTA 0h A</td>\n",
       "      <td>FL-cDNA</td>\n",
       "      <td>RT-PCR</td>\n",
       "      <td>VIRAL RNA</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ABI_SOLID</td>\n",
       "      <td>AB SOLiD System 3.0</td>\n",
       "      <td>SRP002493</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>SRS073299</td>\n",
       "      <td>SAMN00013848</td>\n",
       "      <td>simple</td>\n",
       "      <td>10245</td>\n",
       "      <td>Vaccinia virus</td>\n",
       "      <td>VACV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NIAID-RML-RTS</td>\n",
       "      <td>SRA017695</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>1C9EE0CEFAE2352A0321B24EAB64EE18</td>\n",
       "      <td>753ED95CDA0B112FD877F38960613605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7676059</td>\n",
       "      <td>SRR049675</td>\n",
       "      <td>2010-05-27</td>\n",
       "      <td>2014-05-27 02:58:17</td>\n",
       "      <td>40264881</td>\n",
       "      <td>2013244050</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>1696</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>SRX021028</td>\n",
       "      <td>VVWTA 0.5A</td>\n",
       "      <td>FL-cDNA</td>\n",
       "      <td>RT-PCR</td>\n",
       "      <td>VIRAL RNA</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ABI_SOLID</td>\n",
       "      <td>AB SOLiD System 3.0</td>\n",
       "      <td>SRP002493</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>SRS073299</td>\n",
       "      <td>SAMN00013848</td>\n",
       "      <td>simple</td>\n",
       "      <td>10245</td>\n",
       "      <td>Vaccinia virus</td>\n",
       "      <td>VACV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NIAID-RML-RTS</td>\n",
       "      <td>SRA017695</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>9F96F35BE762574A084985FD9C918C28</td>\n",
       "      <td>392587111A738BC44AE8878E649D052A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7676060</td>\n",
       "      <td>SRR049677</td>\n",
       "      <td>2010-05-27</td>\n",
       "      <td>2014-05-27 02:58:58</td>\n",
       "      <td>40852259</td>\n",
       "      <td>2042612950</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>1732</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>SRX021029</td>\n",
       "      <td>VVWTA 1h A</td>\n",
       "      <td>FL-cDNA</td>\n",
       "      <td>RT-PCR</td>\n",
       "      <td>VIRAL RNA</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ABI_SOLID</td>\n",
       "      <td>AB SOLiD System 3.0</td>\n",
       "      <td>SRP002493</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>SRS073299</td>\n",
       "      <td>SAMN00013848</td>\n",
       "      <td>simple</td>\n",
       "      <td>10245</td>\n",
       "      <td>Vaccinia virus</td>\n",
       "      <td>VACV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NIAID-RML-RTS</td>\n",
       "      <td>SRA017695</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>22DCD0A93DAA207187EF6C6F6A835A55</td>\n",
       "      <td>416EBCA2B1A6AB70E4F778F16C36FCE5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7676061</td>\n",
       "      <td>SRR042496</td>\n",
       "      <td>2010-05-11</td>\n",
       "      <td>2012-01-19 16:52:13</td>\n",
       "      <td>47760</td>\n",
       "      <td>16505980</td>\n",
       "      <td>0</td>\n",
       "      <td>345</td>\n",
       "      <td>42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>SRX020084</td>\n",
       "      <td>VirRnaMgMosqCeI</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>RANDOM PCR</td>\n",
       "      <td>VIRAL RNA</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LS454</td>\n",
       "      <td>454 GS FLX Titanium</td>\n",
       "      <td>SRP002064</td>\n",
       "      <td>PRJNA46359</td>\n",
       "      <td>2.0</td>\n",
       "      <td>46359</td>\n",
       "      <td>SRS011854</td>\n",
       "      <td>SAMN00012272</td>\n",
       "      <td>simple</td>\n",
       "      <td>1284618</td>\n",
       "      <td>environmental samples</td>\n",
       "      <td>VirRnaMgMosqCeI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAN DIEGO STATE UNIVERSITY</td>\n",
       "      <td>SRA012164</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>633E95BCDFC18AD9C8B5DEAAFAE4DD3F</td>\n",
       "      <td>00AB1B11657DAE26FFECDAD623148A3C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7676062</td>\n",
       "      <td>SRR043092</td>\n",
       "      <td>2010-05-10</td>\n",
       "      <td>2012-01-19 16:52:35</td>\n",
       "      <td>162876</td>\n",
       "      <td>43696870</td>\n",
       "      <td>0</td>\n",
       "      <td>268</td>\n",
       "      <td>80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>SRX020242</td>\n",
       "      <td>All HCV</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>PCR</td>\n",
       "      <td>VIRAL RNA</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LS454</td>\n",
       "      <td>454 GS FLX</td>\n",
       "      <td>SRP002438</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>SRS072567</td>\n",
       "      <td>SAMN00012789</td>\n",
       "      <td>simple</td>\n",
       "      <td>11103</td>\n",
       "      <td>Hepacivirus C</td>\n",
       "      <td>all_hcv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UNIVERSITY OF PENNSYLVANIA</td>\n",
       "      <td>SRA012580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>1D70AED9981C7955082992A1AA787246</td>\n",
       "      <td>7C03FE9A87DE427C72136E8C1CA1781C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7676063 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Run ReleaseDate             LoadDate     spots       bases  \\\n",
       "0        ERR3550121  2019-09-25                  NaN         0           0   \n",
       "1        ERR3550142  2019-09-25                  NaN         0           0   \n",
       "2        ERR3550147  2019-09-25                  NaN         0           0   \n",
       "3        ERR3550125  2019-09-25                  NaN         0           0   \n",
       "4        ERR3550137  2019-09-25                  NaN         0           0   \n",
       "...             ...         ...                  ...       ...         ...   \n",
       "7676058   SRR049674  2010-05-27  2014-05-27 02:57:13  35888370  1794418500   \n",
       "7676059   SRR049675  2010-05-27  2014-05-27 02:58:17  40264881  2013244050   \n",
       "7676060   SRR049677  2010-05-27  2014-05-27 02:58:58  40852259  2042612950   \n",
       "7676061   SRR042496  2010-05-11  2012-01-19 16:52:13     47760    16505980   \n",
       "7676062   SRR043092  2010-05-10  2012-01-19 16:52:35    162876    43696870   \n",
       "\n",
       "         spots_with_mates  avgLength  size_MB AssemblyName  \\\n",
       "0                       0          0        0          NaN   \n",
       "1                       0          0        0          NaN   \n",
       "2                       0          0        0          NaN   \n",
       "3                       0          0        0          NaN   \n",
       "4                       0          0        0          NaN   \n",
       "...                   ...        ...      ...          ...   \n",
       "7676058                 0         50     1504          NaN   \n",
       "7676059                 0         50     1696          NaN   \n",
       "7676060                 0         50     1732          NaN   \n",
       "7676061                 0        345       42          NaN   \n",
       "7676062                 0        268       80          NaN   \n",
       "\n",
       "                                             download_path  Experiment  \\\n",
       "0                                                      NaN  ERX3556726   \n",
       "1                                                      NaN  ERX3556747   \n",
       "2                                                      NaN  ERX3556752   \n",
       "3                                                      NaN  ERX3556730   \n",
       "4                                                      NaN  ERX3556742   \n",
       "...                                                    ...         ...   \n",
       "7676058  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...   SRX020916   \n",
       "7676059  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...   SRX021028   \n",
       "7676060  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...   SRX021029   \n",
       "7676061  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...   SRX020084   \n",
       "7676062  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...   SRX020242   \n",
       "\n",
       "             LibraryName LibraryStrategy LibrarySelection LibrarySource  \\\n",
       "0                    NaN             WGS           RANDOM       GENOMIC   \n",
       "1                    NaN             WGS           RANDOM       GENOMIC   \n",
       "2                    NaN             WGS           RANDOM       GENOMIC   \n",
       "3                    NaN             WGS           RANDOM       GENOMIC   \n",
       "4                    NaN             WGS           RANDOM       GENOMIC   \n",
       "...                  ...             ...              ...           ...   \n",
       "7676058       VVWTA 0h A         FL-cDNA           RT-PCR     VIRAL RNA   \n",
       "7676059       VVWTA 0.5A         FL-cDNA           RT-PCR     VIRAL RNA   \n",
       "7676060       VVWTA 1h A         FL-cDNA           RT-PCR     VIRAL RNA   \n",
       "7676061  VirRnaMgMosqCeI           OTHER       RANDOM PCR     VIRAL RNA   \n",
       "7676062          All HCV        AMPLICON              PCR     VIRAL RNA   \n",
       "\n",
       "        LibraryLayout  InsertSize  InsertDev   Platform                Model  \\\n",
       "0              PAIRED         250        0.0   ILLUMINA  Illumina HiSeq 2000   \n",
       "1              PAIRED         250        0.0   ILLUMINA  Illumina HiSeq 2000   \n",
       "2              PAIRED         250        0.0   ILLUMINA  Illumina HiSeq 2000   \n",
       "3              PAIRED         250        0.0   ILLUMINA  Illumina HiSeq 2000   \n",
       "4              PAIRED         250        0.0   ILLUMINA  Illumina HiSeq 2000   \n",
       "...               ...         ...        ...        ...                  ...   \n",
       "7676058        SINGLE           0        0.0  ABI_SOLID  AB SOLiD System 3.0   \n",
       "7676059        SINGLE           0        0.0  ABI_SOLID  AB SOLiD System 3.0   \n",
       "7676060        SINGLE           0        0.0  ABI_SOLID  AB SOLiD System 3.0   \n",
       "7676061        SINGLE           0        0.0      LS454  454 GS FLX Titanium   \n",
       "7676062        SINGLE           0        0.0      LS454           454 GS FLX   \n",
       "\n",
       "          SRAStudy  BioProject Study_Pubmed_id  ProjectID      Sample  \\\n",
       "0        ERP000767   PRJEB2600             NaN     204335  ERS3773255   \n",
       "1        ERP000767   PRJEB2600             NaN     204335  ERS3773255   \n",
       "2        ERP000767   PRJEB2600             NaN     204335  ERS3773263   \n",
       "3        ERP000767   PRJEB2600             NaN     204335  ERS3773263   \n",
       "4        ERP000767   PRJEB2600             NaN     204335  ERS3773252   \n",
       "...            ...         ...             ...        ...         ...   \n",
       "7676058  SRP002493         NaN             2.0          0   SRS073299   \n",
       "7676059  SRP002493         NaN             2.0          0   SRS073299   \n",
       "7676060  SRP002493         NaN             2.0          0   SRS073299   \n",
       "7676061  SRP002064  PRJNA46359             2.0      46359   SRS011854   \n",
       "7676062  SRP002438         NaN             2.0          0   SRS072567   \n",
       "\n",
       "            BioSample SampleType    TaxID         ScientificName  \\\n",
       "0                 NaN     simple        0    Leishmania infantum   \n",
       "1                 NaN     simple        0    Leishmania infantum   \n",
       "2                 NaN     simple        0    Leishmania donovani   \n",
       "3                 NaN     simple        0    Leishmania donovani   \n",
       "4                 NaN     simple        0    Leishmania donovani   \n",
       "...               ...        ...      ...                    ...   \n",
       "7676058  SAMN00013848     simple    10245         Vaccinia virus   \n",
       "7676059  SAMN00013848     simple    10245         Vaccinia virus   \n",
       "7676060  SAMN00013848     simple    10245         Vaccinia virus   \n",
       "7676061  SAMN00012272     simple  1284618  environmental samples   \n",
       "7676062  SAMN00012789     simple    11103          Hepacivirus C   \n",
       "\n",
       "                                   SampleName g1k_pop_code source  \\\n",
       "0        0c4ac31a-dec5-11e9-ba7b-68b599768938          NaN    NaN   \n",
       "1        0c4ac31a-dec5-11e9-ba7b-68b599768938          NaN    NaN   \n",
       "2        0e2d3a2c-dec6-11e9-a5d1-68b59976a384          NaN    NaN   \n",
       "3        0e2d3a2c-dec6-11e9-a5d1-68b59976a384          NaN    NaN   \n",
       "4        188c9a6e-deba-11e9-ba7b-68b599768938          NaN    NaN   \n",
       "...                                       ...          ...    ...   \n",
       "7676058                                  VACV          NaN    NaN   \n",
       "7676059                                  VACV          NaN    NaN   \n",
       "7676060                                  VACV          NaN    NaN   \n",
       "7676061                       VirRnaMgMosqCeI          NaN    NaN   \n",
       "7676062                               all_hcv          NaN    NaN   \n",
       "\n",
       "        g1k_analysis_group Subject_ID  Sex Disease Tumor Affection_Status  \\\n",
       "0                      NaN        NaN  NaN     NaN    no              NaN   \n",
       "1                      NaN        NaN  NaN     NaN    no              NaN   \n",
       "2                      NaN        NaN  NaN     NaN    no              NaN   \n",
       "3                      NaN        NaN  NaN     NaN    no              NaN   \n",
       "4                      NaN        NaN  NaN     NaN    no              NaN   \n",
       "...                    ...        ...  ...     ...   ...              ...   \n",
       "7676058                NaN        NaN  NaN     NaN    no              NaN   \n",
       "7676059                NaN        NaN  NaN     NaN    no              NaN   \n",
       "7676060                NaN        NaN  NaN     NaN    no              NaN   \n",
       "7676061                NaN        NaN  NaN     NaN    no              NaN   \n",
       "7676062                NaN        NaN  NaN     NaN    no              NaN   \n",
       "\n",
       "        Analyte_Type Histological_Type Body_Site                  CenterName  \\\n",
       "0                NaN               NaN       NaN                     UOC-CCG   \n",
       "1                NaN               NaN       NaN                     UOC-CCG   \n",
       "2                NaN               NaN       NaN                     UOC-CCG   \n",
       "3                NaN               NaN       NaN                     UOC-CCG   \n",
       "4                NaN               NaN       NaN                     UOC-CCG   \n",
       "...              ...               ...       ...                         ...   \n",
       "7676058          NaN               NaN       NaN               NIAID-RML-RTS   \n",
       "7676059          NaN               NaN       NaN               NIAID-RML-RTS   \n",
       "7676060          NaN               NaN       NaN               NIAID-RML-RTS   \n",
       "7676061          NaN               NaN       NaN  SAN DIEGO STATE UNIVERSITY   \n",
       "7676062          NaN               NaN       NaN  UNIVERSITY OF PENNSYLVANIA   \n",
       "\n",
       "         Submission dbgap_study_accession Consent  \\\n",
       "0        ERA2143351                   NaN  public   \n",
       "1        ERA2143351                   NaN  public   \n",
       "2        ERA2143351                   NaN  public   \n",
       "3        ERA2143351                   NaN  public   \n",
       "4        ERA2143351                   NaN  public   \n",
       "...             ...                   ...     ...   \n",
       "7676058   SRA017695                   NaN  public   \n",
       "7676059   SRA017695                   NaN  public   \n",
       "7676060   SRA017695                   NaN  public   \n",
       "7676061   SRA012164                   NaN  public   \n",
       "7676062   SRA012580                   NaN  public   \n",
       "\n",
       "                                  RunHash                          ReadHash  \n",
       "0                                     NaN                               NaN  \n",
       "1                                     NaN                               NaN  \n",
       "2                                     NaN                               NaN  \n",
       "3                                     NaN                               NaN  \n",
       "4                                     NaN                               NaN  \n",
       "...                                   ...                               ...  \n",
       "7676058  1C9EE0CEFAE2352A0321B24EAB64EE18  753ED95CDA0B112FD877F38960613605  \n",
       "7676059  9F96F35BE762574A084985FD9C918C28  392587111A738BC44AE8878E649D052A  \n",
       "7676060  22DCD0A93DAA207187EF6C6F6A835A55  416EBCA2B1A6AB70E4F778F16C36FCE5  \n",
       "7676061  633E95BCDFC18AD9C8B5DEAAFAE4DD3F  00AB1B11657DAE26FFECDAD623148A3C  \n",
       "7676062  1D70AED9981C7955082992A1AA787246  7C03FE9A87DE427C72136E8C1CA1781C  \n",
       "\n",
       "[7676063 rows x 47 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Convert dates to a universal format by ignoring time of day\n",
    "pd.set_option('display.max_columns', 50)\n",
    "sraAttributes['ReleaseDate'] = sraAttributes['ReleaseDate'].str[0:10]\n",
    "sraAttributes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import GEO attribute data and add Series column\n",
    "geoAttributes = pd.read_csv('../geo_platforms_table.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Platform</th>\n",
       "      <th>Title</th>\n",
       "      <th>Technology</th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>Data Rows</th>\n",
       "      <th>Samples Count</th>\n",
       "      <th>Series Count</th>\n",
       "      <th>Contact</th>\n",
       "      <th>Release Date</th>\n",
       "      <th>Series</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Datasets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>GPL4</td>\n",
       "      <td>SAGE:10:NlaIII:Homo sapiens</td>\n",
       "      <td>SAGE NlaIII</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>265577</td>\n",
       "      <td>580</td>\n",
       "      <td>58</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2000-09-28</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>GPL4</td>\n",
       "      <td>SAGE:10:NlaIII:Homo sapiens</td>\n",
       "      <td>SAGE NlaIII</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>265577</td>\n",
       "      <td>580</td>\n",
       "      <td>58</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2000-09-28</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>GSM2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>GPL4</td>\n",
       "      <td>SAGE:10:NlaIII:Homo sapiens</td>\n",
       "      <td>SAGE NlaIII</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>265577</td>\n",
       "      <td>580</td>\n",
       "      <td>58</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2000-09-28</td>\n",
       "      <td>GSE10</td>\n",
       "      <td>GSM571</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>GPL4</td>\n",
       "      <td>SAGE:10:NlaIII:Homo sapiens</td>\n",
       "      <td>SAGE NlaIII</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>265577</td>\n",
       "      <td>580</td>\n",
       "      <td>58</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2000-09-28</td>\n",
       "      <td>GSE10</td>\n",
       "      <td>GSM572</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>GPL4</td>\n",
       "      <td>SAGE:10:NlaIII:Homo sapiens</td>\n",
       "      <td>SAGE NlaIII</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>265577</td>\n",
       "      <td>580</td>\n",
       "      <td>58</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2000-09-28</td>\n",
       "      <td>GSE10</td>\n",
       "      <td>GSM573</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2961228</td>\n",
       "      <td>GPL27549</td>\n",
       "      <td>Illumina HiSeq 2000 (Escherichia coli str. K-1...</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>Escherichia coli str. K-12 substr. DH10B</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-09-30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2961229</td>\n",
       "      <td>GPL27550</td>\n",
       "      <td>Illumina HiSeq 4000 (Solanum chacoense)</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>Solanum chacoense</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-09-30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2961230</td>\n",
       "      <td>GPL27552</td>\n",
       "      <td>Illumina HiSeq 2500 (Caenorhabditis elegans ,V...</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>Caenorhabditis elegans ,Vibrio parahaemolyticus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-09-30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2961231</td>\n",
       "      <td>GPL27554</td>\n",
       "      <td>Illumina HiSeq 2000 (Trichoderma atroviride)</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>Trichoderma atroviride</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-09-30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2961232</td>\n",
       "      <td>GPL27555</td>\n",
       "      <td>Illumina HiSeq 2500 (Caenorhabditis elegans; V...</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>Caenorhabditis elegans;Vibrio parahaemolyticus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-09-30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2961233 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Platform                                              Title  \\\n",
       "0            GPL4                        SAGE:10:NlaIII:Homo sapiens   \n",
       "1            GPL4                        SAGE:10:NlaIII:Homo sapiens   \n",
       "2            GPL4                        SAGE:10:NlaIII:Homo sapiens   \n",
       "3            GPL4                        SAGE:10:NlaIII:Homo sapiens   \n",
       "4            GPL4                        SAGE:10:NlaIII:Homo sapiens   \n",
       "...           ...                                                ...   \n",
       "2961228  GPL27549  Illumina HiSeq 2000 (Escherichia coli str. K-1...   \n",
       "2961229  GPL27550            Illumina HiSeq 4000 (Solanum chacoense)   \n",
       "2961230  GPL27552  Illumina HiSeq 2500 (Caenorhabditis elegans ,V...   \n",
       "2961231  GPL27554       Illumina HiSeq 2000 (Trichoderma atroviride)   \n",
       "2961232  GPL27555  Illumina HiSeq 2500 (Caenorhabditis elegans; V...   \n",
       "\n",
       "                         Technology  \\\n",
       "0                       SAGE NlaIII   \n",
       "1                       SAGE NlaIII   \n",
       "2                       SAGE NlaIII   \n",
       "3                       SAGE NlaIII   \n",
       "4                       SAGE NlaIII   \n",
       "...                             ...   \n",
       "2961228  high-throughput sequencing   \n",
       "2961229  high-throughput sequencing   \n",
       "2961230  high-throughput sequencing   \n",
       "2961231  high-throughput sequencing   \n",
       "2961232  high-throughput sequencing   \n",
       "\n",
       "                                                Taxonomy  Data Rows  \\\n",
       "0                                           Homo sapiens     265577   \n",
       "1                                           Homo sapiens     265577   \n",
       "2                                           Homo sapiens     265577   \n",
       "3                                           Homo sapiens     265577   \n",
       "4                                           Homo sapiens     265577   \n",
       "...                                                  ...        ...   \n",
       "2961228         Escherichia coli str. K-12 substr. DH10B          0   \n",
       "2961229                                Solanum chacoense          0   \n",
       "2961230  Caenorhabditis elegans ,Vibrio parahaemolyticus          0   \n",
       "2961231                           Trichoderma atroviride          0   \n",
       "2961232   Caenorhabditis elegans;Vibrio parahaemolyticus          0   \n",
       "\n",
       "         Samples Count  Series Count Contact Release Date  Series Accession  \\\n",
       "0                  580            58     GEO   2000-09-28  GSE506      GSM1   \n",
       "1                  580            58     GEO   2000-09-28  GSE506      GSM2   \n",
       "2                  580            58     GEO   2000-09-28   GSE10    GSM571   \n",
       "3                  580            58     GEO   2000-09-28   GSE10    GSM572   \n",
       "4                  580            58     GEO   2000-09-28   GSE10    GSM573   \n",
       "...                ...           ...     ...          ...     ...       ...   \n",
       "2961228              0             0     GEO   2019-09-30     NaN       NaN   \n",
       "2961229              0             0     GEO   2019-09-30     NaN       NaN   \n",
       "2961230              0             0     GEO   2019-09-30     NaN       NaN   \n",
       "2961231              0             0     GEO   2019-09-30     NaN       NaN   \n",
       "2961232              0             0     GEO   2019-09-30     NaN       NaN   \n",
       "\n",
       "        Datasets  \n",
       "0            NaN  \n",
       "1            NaN  \n",
       "2            NaN  \n",
       "3            NaN  \n",
       "4            NaN  \n",
       "...          ...  \n",
       "2961228      NaN  \n",
       "2961229      NaN  \n",
       "2961230      NaN  \n",
       "2961231      NaN  \n",
       "2961232      NaN  \n",
       "\n",
       "[2961233 rows x 12 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "geoAttributes.rename(columns={'Accession':'Platform'}, inplace=True)\n",
    "geoAttributes = pd.merge(geoAttributes, geoReference, how = 'left', on = 'Platform')\n",
    "dates = []\n",
    "for i in geoAttributes['Release Date']:\n",
    "    dates.append(convGEODate(i))\n",
    "geoAttributes['Release Date'] = dates\n",
    "geoAttributes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/niko/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  app.launch_new_instance()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>pmc_ID</th>\n",
       "      <th>accession</th>\n",
       "      <th>converted_accession</th>\n",
       "      <th>repository</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3707052</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>GEO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3706879</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GEO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4255636</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GEO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GEO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GEO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78614</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589307</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GEO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78615</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589301</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GEO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78616</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1059</td>\n",
       "      <td>GSE2191</td>\n",
       "      <td>GEO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78618</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>PRJNA437812</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78619</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73490 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   journal      pmc_ID    accession converted_accession  \\\n",
       "0      Alzheimers_Res_Ther  PMC3707052         GSM1              GSE506   \n",
       "1      Alzheimers_Res_Ther  PMC3706879     GSE45534            GSE45534   \n",
       "2      Alzheimers_Res_Ther  PMC4255636      GSE5281             GSE5281   \n",
       "3      Alzheimers_Res_Ther  PMC4731966     GSE67036            GSE67036   \n",
       "4      Alzheimers_Res_Ther  PMC4731966      GSE1297             GSE1297   \n",
       "...                    ...         ...          ...                 ...   \n",
       "78614          Dis_Markers  PMC6589307     GSE14795            GSE14795   \n",
       "78615          Dis_Markers  PMC6589301     GSE62254            GSE62254   \n",
       "78616          Dis_Markers  PMC3834650      GDS1059             GSE2191   \n",
       "78618     Case_Rep_Hematol  PMC6462343  PRJNA437812           SRP135146   \n",
       "78619     Case_Rep_Hematol  PMC6462343    SRP135146           SRP135146   \n",
       "\n",
       "      repository  \n",
       "0            GEO  \n",
       "1            GEO  \n",
       "2            GEO  \n",
       "3            GEO  \n",
       "4            GEO  \n",
       "...          ...  \n",
       "78614        GEO  \n",
       "78615        GEO  \n",
       "78616        GEO  \n",
       "78618        SRA  \n",
       "78619        SRA  \n",
       "\n",
       "[73490 rows x 5 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Add a column tagging each accession as GEO or SRA\n",
    "\n",
    "repoList = []\n",
    "\n",
    "for i in pmcData['converted_accession']:\n",
    "    if(type(i) == str):\n",
    "        if('GSE' in i):\n",
    "            repoList.append('GEO')\n",
    "        elif('SRP' in i or 'ERP' in i or 'DRP' in i):\n",
    "            repoList.append('SRA')\n",
    "        else:\n",
    "            repoList.append(np.NaN)\n",
    "    else:\n",
    "        repoList.append(np.NaN)\n",
    "        \n",
    "pmcData['repository'] = repoList\n",
    "pmcData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>pmc_ID</th>\n",
       "      <th>accession</th>\n",
       "      <th>converted_accession</th>\n",
       "      <th>repository</th>\n",
       "      <th>pmc_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3707052</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-4-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3706879</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-5-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4255636</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2014-11-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2016-1-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2016-1-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73485</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589307</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-6-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73486</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589301</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-6-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73487</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1059</td>\n",
       "      <td>GSE2191</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-11-5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73488</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>PRJNA437812</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRA</td>\n",
       "      <td>2019-3-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73489</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRA</td>\n",
       "      <td>2019-3-31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73490 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   journal      pmc_ID    accession converted_accession  \\\n",
       "0      Alzheimers_Res_Ther  PMC3707052         GSM1              GSE506   \n",
       "1      Alzheimers_Res_Ther  PMC3706879     GSE45534            GSE45534   \n",
       "2      Alzheimers_Res_Ther  PMC4255636      GSE5281             GSE5281   \n",
       "3      Alzheimers_Res_Ther  PMC4731966     GSE67036            GSE67036   \n",
       "4      Alzheimers_Res_Ther  PMC4731966      GSE1297             GSE1297   \n",
       "...                    ...         ...          ...                 ...   \n",
       "73485          Dis_Markers  PMC6589307     GSE14795            GSE14795   \n",
       "73486          Dis_Markers  PMC6589301     GSE62254            GSE62254   \n",
       "73487          Dis_Markers  PMC3834650      GDS1059             GSE2191   \n",
       "73488     Case_Rep_Hematol  PMC6462343  PRJNA437812           SRP135146   \n",
       "73489     Case_Rep_Hematol  PMC6462343    SRP135146           SRP135146   \n",
       "\n",
       "      repository   pmc_date  \n",
       "0            GEO  2013-4-18  \n",
       "1            GEO  2013-5-14  \n",
       "2            GEO  2014-11-2  \n",
       "3            GEO  2016-1-28  \n",
       "4            GEO  2016-1-28  \n",
       "...          ...        ...  \n",
       "73485        GEO   2019-6-4  \n",
       "73486        GEO   2019-6-2  \n",
       "73487        GEO  2013-11-5  \n",
       "73488        SRA  2019-3-31  \n",
       "73489        SRA  2019-3-31  \n",
       "\n",
       "[73490 rows x 6 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add column for paper publish date\n",
    "# this data was scraped from XML files on the hoffman2 cluster: /u/scratch/n/nikodm/pmcOA/\n",
    "\n",
    "pmc_dates = pd.read_csv('../data_lists/postFilterDates.csv').drop_duplicates()\n",
    "\n",
    "pmcData = pd.merge(pmcData, pmc_dates, how = 'left', on = 'pmc_ID')\n",
    "pmcData = pmcData.rename(columns = {'date': 'pmc_date'})\n",
    "\n",
    "pmcData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get every factor we're interested in from our tables of GEO and SRA metadata...\n",
    "\n",
    "# take a slice of the GEO and SRA attribute tables with only the info we want\n",
    "slicedGEOAtt = geoAttributes[['Series', 'Release Date', 'Technology', 'Taxonomy']]\n",
    "slicedGEOAtt.columns = ['converted_accession', 'geoRelease', 'geoHardware', 'geoSpecies']\n",
    "slicedGEOAtt = slicedGEOAtt.drop_duplicates(subset = ['converted_accession'])\n",
    "\n",
    "slicedSRAAtt = sraAttributes[['SRAStudy', 'ReleaseDate', 'Model', \n",
    "                              'LibraryStrategy', 'ScientificName', \n",
    "                              'bases', 'avgLength', 'Consent']]\n",
    "slicedSRAAtt.columns = ['converted_accession', 'sraRelease', 'sraHardware', \n",
    "                        'sraLibrary_strategy', 'sraSpecies', \n",
    "                        'sraBases', 'sraAvg_length', 'sraAccess']\n",
    "slicedSRAAtt = slicedSRAAtt.drop_duplicates(subset = ['converted_accession'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hardware</th>\n",
       "      <th>geoLibrary_strategy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>SAGE NlaIII</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>spotted DNA/cDNA</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>SAGE Sau3A</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>spotted oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>antibody</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>MS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>SARST</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>SAGE RsaI</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>MPSS</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>oligonucleotide beads</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>RT-PCR</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>mixed spotted oligonucleotide/cDNA</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>spotted peptide or protein</td>\n",
       "      <td>Expression_Array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>RNA-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>tissue</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              hardware geoLibrary_strategy\n",
       "0                          SAGE NlaIII    Expression_Array\n",
       "1                     spotted DNA/cDNA    Expression_Array\n",
       "2                           SAGE Sau3A    Expression_Array\n",
       "3              in situ oligonucleotide    Expression_Array\n",
       "4              spotted oligonucleotide    Expression_Array\n",
       "5                             antibody    Expression_Array\n",
       "6                                   MS                 NaN\n",
       "7                                SARST                 NaN\n",
       "8                            SAGE RsaI                 NaN\n",
       "9                                other                 NaN\n",
       "10                                MPSS    Expression_Array\n",
       "11               oligonucleotide beads    Expression_Array\n",
       "12                              RT-PCR    Expression_Array\n",
       "13  mixed spotted oligonucleotide/cDNA    Expression_Array\n",
       "14          spotted peptide or protein    Expression_Array\n",
       "15                                 NaN                 NaN\n",
       "16          high-throughput sequencing             RNA-Seq\n",
       "17                              tissue                 NaN"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# special case for GEO: make an educated guess on library strategy based on hardware\n",
    "# These guesses are based on manually checking GEO series IDs that corresponded to various types of hardware\n",
    "\n",
    "gc = Counter(geoAttributes['Technology'])\n",
    "ls_guesses = pd.DataFrame.from_dict(gc, orient='index').reset_index()\n",
    "ls_guesses.columns = ['hardware', 'use_count']\n",
    "ls_guesses = ls_guesses.drop(labels = ['use_count'], axis = 1)\n",
    "\n",
    "ls = []\n",
    "\n",
    "for i in ls_guesses['hardware']:\n",
    "    if(i == 'high-throughput sequencing'):\n",
    "        ls.append('RNA-Seq')\n",
    "    elif(i == 'SAGE NlaIII' or i == 'spotted DNA/cDNA' or i == 'SAGE Sau3A' \n",
    "         or i == 'in situ oligonucleotide' or i == 'spotted oligonucleotide' \n",
    "         or i == 'antibody' or i == 'MPSS' or i == 'oligonucleotide beads' \n",
    "         or i == 'RT-PCR' or i == 'mixed spotted oligonucleotide/cDNA' \n",
    "         or i == 'spotted peptide or protein'):\n",
    "        ls.append('Expression_Array')\n",
    "    else:\n",
    "        ls.append(np.NaN)\n",
    "        \n",
    "ls_guesses.loc[:,'geoLibrary_strategy'] = ls\n",
    "ls_guesses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge SRA attributes onto pmcData table\n",
    "mergedSRA = pd.merge(pmcData, slicedSRAAtt, how = 'left', on = 'converted_accession')\n",
    "mergedSRA = mergedSRA.drop_duplicates()\n",
    "\n",
    "# merge GEO attributes onto table of pmcData + SRA Attributes\n",
    "allFactors = pd.merge(mergedSRA, slicedGEOAtt, how = 'left', on = 'converted_accession')\n",
    "allFactors = pd.merge(allFactors, ls_guesses, how = 'left', left_on = 'geoHardware', right_on = 'hardware')\n",
    "\n",
    "allFactors = allFactors.dropna(subset = ['converted_accession'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>pmc_ID</th>\n",
       "      <th>accession</th>\n",
       "      <th>converted_accession</th>\n",
       "      <th>repository</th>\n",
       "      <th>pmc_date</th>\n",
       "      <th>repository_date</th>\n",
       "      <th>species</th>\n",
       "      <th>hardware</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>sraAvg_length</th>\n",
       "      <th>sraBases</th>\n",
       "      <th>sraAccess</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3707052</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-4-18</td>\n",
       "      <td>2000-09-28</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>SAGE NlaIII</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3706879</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-5-14</td>\n",
       "      <td>2004-05-25</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4255636</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2014-11-2</td>\n",
       "      <td>2003-11-07</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2016-1-28</td>\n",
       "      <td>2013-05-06</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2016-1-28</td>\n",
       "      <td>2002-03-11</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73485</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589307</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-6-4</td>\n",
       "      <td>2002-03-11</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73486</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589301</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-6-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73487</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1059</td>\n",
       "      <td>GSE2191</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-11-5</td>\n",
       "      <td>2009-03-16</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73488</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>PRJNA437812</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRA</td>\n",
       "      <td>2019-3-31</td>\n",
       "      <td>2018-03-13</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>602.0</td>\n",
       "      <td>17743348.0</td>\n",
       "      <td>public</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73489</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRA</td>\n",
       "      <td>2019-3-31</td>\n",
       "      <td>2018-03-13</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>602.0</td>\n",
       "      <td>17743348.0</td>\n",
       "      <td>public</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73490 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   journal      pmc_ID    accession converted_accession  \\\n",
       "0      Alzheimers_Res_Ther  PMC3707052         GSM1              GSE506   \n",
       "1      Alzheimers_Res_Ther  PMC3706879     GSE45534            GSE45534   \n",
       "2      Alzheimers_Res_Ther  PMC4255636      GSE5281             GSE5281   \n",
       "3      Alzheimers_Res_Ther  PMC4731966     GSE67036            GSE67036   \n",
       "4      Alzheimers_Res_Ther  PMC4731966      GSE1297             GSE1297   \n",
       "...                    ...         ...          ...                 ...   \n",
       "73485          Dis_Markers  PMC6589307     GSE14795            GSE14795   \n",
       "73486          Dis_Markers  PMC6589301     GSE62254            GSE62254   \n",
       "73487          Dis_Markers  PMC3834650      GDS1059             GSE2191   \n",
       "73488     Case_Rep_Hematol  PMC6462343  PRJNA437812           SRP135146   \n",
       "73489     Case_Rep_Hematol  PMC6462343    SRP135146           SRP135146   \n",
       "\n",
       "      repository   pmc_date repository_date               species  \\\n",
       "0            GEO  2013-4-18      2000-09-28          Homo sapiens   \n",
       "1            GEO  2013-5-14      2004-05-25          Mus musculus   \n",
       "2            GEO  2014-11-2      2003-11-07          Homo sapiens   \n",
       "3            GEO  2016-1-28      2013-05-06     Rattus norvegicus   \n",
       "4            GEO  2016-1-28      2002-03-11          Homo sapiens   \n",
       "...          ...        ...             ...                   ...   \n",
       "73485        GEO   2019-6-4      2002-03-11          Homo sapiens   \n",
       "73486        GEO   2019-6-2             NaN                   NaN   \n",
       "73487        GEO  2013-11-5      2009-03-16          Homo sapiens   \n",
       "73488        SRA  2019-3-31      2018-03-13  human gut metagenome   \n",
       "73489        SRA  2019-3-31      2018-03-13  human gut metagenome   \n",
       "\n",
       "                      hardware  library_strategy  sraAvg_length    sraBases  \\\n",
       "0                  SAGE NlaIII  Expression_Array            NaN         NaN   \n",
       "1      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "2      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "3      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "4      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "...                        ...               ...            ...         ...   \n",
       "73485  in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "73486                      NaN               NaN            NaN         NaN   \n",
       "73487  in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "73488           Illumina MiSeq          AMPLICON          602.0  17743348.0   \n",
       "73489           Illumina MiSeq          AMPLICON          602.0  17743348.0   \n",
       "\n",
       "      sraAccess  \n",
       "0           NaN  \n",
       "1           NaN  \n",
       "2           NaN  \n",
       "3           NaN  \n",
       "4           NaN  \n",
       "...         ...  \n",
       "73485       NaN  \n",
       "73486       NaN  \n",
       "73487       NaN  \n",
       "73488    public  \n",
       "73489    public  \n",
       "\n",
       "[73490 rows x 13 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# clean up columns with factor for both SRA and GEO, rearrange columns\n",
    "allFactors['species'] = allFactors['sraSpecies'].fillna(allFactors['geoSpecies'])\n",
    "allFactors = allFactors.drop(labels = ['sraSpecies', 'geoSpecies'], axis = 1)\n",
    "\n",
    "allFactors['hardware'] = allFactors['sraHardware'].fillna(allFactors['geoHardware'])\n",
    "allFactors = allFactors.drop(labels = ['sraHardware', 'geoHardware'], axis = 1)\n",
    "\n",
    "allFactors['library_strategy'] = allFactors['sraLibrary_strategy'].fillna(allFactors['geoLibrary_strategy'])\n",
    "allFactors = allFactors.drop(labels = ['sraLibrary_strategy', 'geoLibrary_strategy'], axis = 1)\n",
    "\n",
    "allFactors['repository_date'] = allFactors['sraRelease'].fillna(allFactors['geoRelease'])\n",
    "allFactors = allFactors.drop(labels = ['sraRelease', 'geoRelease'], axis = 1)\n",
    "\n",
    "cols = ['journal', 'pmc_ID', 'accession', 'converted_accession', 'repository', \n",
    "        'pmc_date', 'repository_date', 'species', \n",
    "        'hardware', 'library_strategy', 'sraAvg_length', 'sraBases', 'sraAccess']\n",
    "\n",
    "allFactors = allFactors[cols]\n",
    "allFactors"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Step III: Conduct analysis on this table, producing new metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>converted_accession</th>\n",
       "      <th>total_use_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35221</td>\n",
       "      <td>GSE85841</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35222</td>\n",
       "      <td>GPL20115</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35223</td>\n",
       "      <td>GSE73953</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35224</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35225</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35226 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      converted_accession  total_use_count\n",
       "0                  GSE506               64\n",
       "1                GSE45534                1\n",
       "2                 GSE5281               43\n",
       "3                GSE67036                1\n",
       "4                 GSE1297               32\n",
       "...                   ...              ...\n",
       "35221            GSE85841                1\n",
       "35222            GPL20115                1\n",
       "35223            GSE73953                1\n",
       "35224            GSE14795                1\n",
       "35225           SRP135146                1\n",
       "\n",
       "[35226 rows x 2 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add a column with the total use count of each converted_accession ID\n",
    "\n",
    "# first, create a new dataframe with no double-counted accessions\n",
    "# justAcc includes all unique pairings of PMC ID and Series/Study accession\n",
    "justAcc = allFactors.loc[:, ['pmc_ID', 'converted_accession']]\n",
    "justAcc = justAcc.drop_duplicates()\n",
    "\n",
    "rc = Counter(justAcc['converted_accession'])\n",
    "reuse_counts = pd.DataFrame.from_dict(rc, orient = 'index').reset_index()\n",
    "reuse_counts.columns = ['converted_accession', 'total_use_count']\n",
    "reuse_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>pmc_ID</th>\n",
       "      <th>accession</th>\n",
       "      <th>converted_accession</th>\n",
       "      <th>repository</th>\n",
       "      <th>pmc_date</th>\n",
       "      <th>repository_date</th>\n",
       "      <th>species</th>\n",
       "      <th>hardware</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>sraAvg_length</th>\n",
       "      <th>sraBases</th>\n",
       "      <th>sraAccess</th>\n",
       "      <th>total_use_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3707052</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-4-18</td>\n",
       "      <td>2000-09-28</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>SAGE NlaIII</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3706879</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-5-14</td>\n",
       "      <td>2004-05-25</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4255636</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2014-11-2</td>\n",
       "      <td>2003-11-07</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2016-1-28</td>\n",
       "      <td>2013-05-06</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2016-1-28</td>\n",
       "      <td>2002-03-11</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73485</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589307</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-6-4</td>\n",
       "      <td>2002-03-11</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73486</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589301</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-6-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73487</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1059</td>\n",
       "      <td>GSE2191</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-11-5</td>\n",
       "      <td>2009-03-16</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73488</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>PRJNA437812</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRA</td>\n",
       "      <td>2019-3-31</td>\n",
       "      <td>2018-03-13</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>602.0</td>\n",
       "      <td>17743348.0</td>\n",
       "      <td>public</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73489</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRA</td>\n",
       "      <td>2019-3-31</td>\n",
       "      <td>2018-03-13</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>602.0</td>\n",
       "      <td>17743348.0</td>\n",
       "      <td>public</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73490 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   journal      pmc_ID    accession converted_accession  \\\n",
       "0      Alzheimers_Res_Ther  PMC3707052         GSM1              GSE506   \n",
       "1      Alzheimers_Res_Ther  PMC3706879     GSE45534            GSE45534   \n",
       "2      Alzheimers_Res_Ther  PMC4255636      GSE5281             GSE5281   \n",
       "3      Alzheimers_Res_Ther  PMC4731966     GSE67036            GSE67036   \n",
       "4      Alzheimers_Res_Ther  PMC4731966      GSE1297             GSE1297   \n",
       "...                    ...         ...          ...                 ...   \n",
       "73485          Dis_Markers  PMC6589307     GSE14795            GSE14795   \n",
       "73486          Dis_Markers  PMC6589301     GSE62254            GSE62254   \n",
       "73487          Dis_Markers  PMC3834650      GDS1059             GSE2191   \n",
       "73488     Case_Rep_Hematol  PMC6462343  PRJNA437812           SRP135146   \n",
       "73489     Case_Rep_Hematol  PMC6462343    SRP135146           SRP135146   \n",
       "\n",
       "      repository   pmc_date repository_date               species  \\\n",
       "0            GEO  2013-4-18      2000-09-28          Homo sapiens   \n",
       "1            GEO  2013-5-14      2004-05-25          Mus musculus   \n",
       "2            GEO  2014-11-2      2003-11-07          Homo sapiens   \n",
       "3            GEO  2016-1-28      2013-05-06     Rattus norvegicus   \n",
       "4            GEO  2016-1-28      2002-03-11          Homo sapiens   \n",
       "...          ...        ...             ...                   ...   \n",
       "73485        GEO   2019-6-4      2002-03-11          Homo sapiens   \n",
       "73486        GEO   2019-6-2             NaN                   NaN   \n",
       "73487        GEO  2013-11-5      2009-03-16          Homo sapiens   \n",
       "73488        SRA  2019-3-31      2018-03-13  human gut metagenome   \n",
       "73489        SRA  2019-3-31      2018-03-13  human gut metagenome   \n",
       "\n",
       "                      hardware  library_strategy  sraAvg_length    sraBases  \\\n",
       "0                  SAGE NlaIII  Expression_Array            NaN         NaN   \n",
       "1      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "2      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "3      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "4      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "...                        ...               ...            ...         ...   \n",
       "73485  in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "73486                      NaN               NaN            NaN         NaN   \n",
       "73487  in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "73488           Illumina MiSeq          AMPLICON          602.0  17743348.0   \n",
       "73489           Illumina MiSeq          AMPLICON          602.0  17743348.0   \n",
       "\n",
       "      sraAccess  total_use_count  \n",
       "0           NaN               64  \n",
       "1           NaN                1  \n",
       "2           NaN               43  \n",
       "3           NaN                1  \n",
       "4           NaN               32  \n",
       "...         ...              ...  \n",
       "73485       NaN                1  \n",
       "73486       NaN               22  \n",
       "73487       NaN                5  \n",
       "73488    public                1  \n",
       "73489    public                1  \n",
       "\n",
       "[73490 rows x 14 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge reuse counts onto table\n",
    "allFactors = pd.merge(allFactors, reuse_counts, how = 'left', on = 'converted_accession')\n",
    "allFactors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>converted_accession</th>\n",
       "      <th>min_pmc_date</th>\n",
       "      <th>reuse_role</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>DRP000010</td>\n",
       "      <td>2010-4-27</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DRP000017</td>\n",
       "      <td>2011-10-9</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>DRP000018</td>\n",
       "      <td>2011-10-9</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>DRP000019</td>\n",
       "      <td>2011-10-9</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DRP000020</td>\n",
       "      <td>2011-10-9</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35221</td>\n",
       "      <td>SRP212249</td>\n",
       "      <td>2019-7-08</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35222</td>\n",
       "      <td>SRP212328</td>\n",
       "      <td>2019-7-03</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35223</td>\n",
       "      <td>SRP216761</td>\n",
       "      <td>2017-11-21</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35224</td>\n",
       "      <td>SRP221037</td>\n",
       "      <td>2019-3-21</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35225</td>\n",
       "      <td>SRP223031</td>\n",
       "      <td>2018-10-16</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35226 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      converted_accession min_pmc_date reuse_role\n",
       "0               DRP000010    2010-4-27          G\n",
       "1               DRP000017    2011-10-9          G\n",
       "2               DRP000018    2011-10-9          G\n",
       "3               DRP000019    2011-10-9          G\n",
       "4               DRP000020    2011-10-9          G\n",
       "...                   ...          ...        ...\n",
       "35221           SRP212249    2019-7-08          G\n",
       "35222           SRP212328    2019-7-03          G\n",
       "35223           SRP216761   2017-11-21          G\n",
       "35224           SRP221037    2019-3-21          G\n",
       "35225           SRP223031   2018-10-16          G\n",
       "\n",
       "[35226 rows x 3 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Tag each paper as either a generator or a reuser\n",
    "\n",
    "# take each converted accession along with the first date it appears in a paper\n",
    "justPap = allFactors.loc[:, ['converted_accession', 'pmc_date']].drop_duplicates()\n",
    "justPap = justPap.groupby('converted_accession', as_index = False).agg({'pmc_date': 'min'})\n",
    "justPap = justPap.rename(columns = {'pmc_date':'min_pmc_date'})\n",
    "\n",
    "# label all of the usages of these datasets on these dates with a 'G' (generators)\n",
    "justPap.loc[:,'reuse_role'] = 'G'\n",
    "justPap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>pmc_ID</th>\n",
       "      <th>accession</th>\n",
       "      <th>converted_accession</th>\n",
       "      <th>repository</th>\n",
       "      <th>pmc_date</th>\n",
       "      <th>repository_date</th>\n",
       "      <th>species</th>\n",
       "      <th>hardware</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>sraAvg_length</th>\n",
       "      <th>sraBases</th>\n",
       "      <th>sraAccess</th>\n",
       "      <th>total_use_count</th>\n",
       "      <th>reuse_role</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3707052</td>\n",
       "      <td>GSM1</td>\n",
       "      <td>GSE506</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-4-18</td>\n",
       "      <td>2000-09-28</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>SAGE NlaIII</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64</td>\n",
       "      <td>R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC3706879</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GSE45534</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-5-14</td>\n",
       "      <td>2004-05-25</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4255636</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GSE5281</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2014-11-2</td>\n",
       "      <td>2003-11-07</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>43</td>\n",
       "      <td>R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GSE67036</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2016-1-28</td>\n",
       "      <td>2013-05-06</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Alzheimers_Res_Ther</td>\n",
       "      <td>PMC4731966</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GSE1297</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2016-1-28</td>\n",
       "      <td>2002-03-11</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32</td>\n",
       "      <td>R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73485</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589307</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GSE14795</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-6-4</td>\n",
       "      <td>2002-03-11</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73486</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC6589301</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GSE62254</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2019-6-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22</td>\n",
       "      <td>R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73487</td>\n",
       "      <td>Dis_Markers</td>\n",
       "      <td>PMC3834650</td>\n",
       "      <td>GDS1059</td>\n",
       "      <td>GSE2191</td>\n",
       "      <td>GEO</td>\n",
       "      <td>2013-11-5</td>\n",
       "      <td>2009-03-16</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>Expression_Array</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73488</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>PRJNA437812</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRA</td>\n",
       "      <td>2019-3-31</td>\n",
       "      <td>2018-03-13</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>602.0</td>\n",
       "      <td>17743348.0</td>\n",
       "      <td>public</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73489</td>\n",
       "      <td>Case_Rep_Hematol</td>\n",
       "      <td>PMC6462343</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRP135146</td>\n",
       "      <td>SRA</td>\n",
       "      <td>2019-3-31</td>\n",
       "      <td>2018-03-13</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>602.0</td>\n",
       "      <td>17743348.0</td>\n",
       "      <td>public</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73490 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   journal      pmc_ID    accession converted_accession  \\\n",
       "0      Alzheimers_Res_Ther  PMC3707052         GSM1              GSE506   \n",
       "1      Alzheimers_Res_Ther  PMC3706879     GSE45534            GSE45534   \n",
       "2      Alzheimers_Res_Ther  PMC4255636      GSE5281             GSE5281   \n",
       "3      Alzheimers_Res_Ther  PMC4731966     GSE67036            GSE67036   \n",
       "4      Alzheimers_Res_Ther  PMC4731966      GSE1297             GSE1297   \n",
       "...                    ...         ...          ...                 ...   \n",
       "73485          Dis_Markers  PMC6589307     GSE14795            GSE14795   \n",
       "73486          Dis_Markers  PMC6589301     GSE62254            GSE62254   \n",
       "73487          Dis_Markers  PMC3834650      GDS1059             GSE2191   \n",
       "73488     Case_Rep_Hematol  PMC6462343  PRJNA437812           SRP135146   \n",
       "73489     Case_Rep_Hematol  PMC6462343    SRP135146           SRP135146   \n",
       "\n",
       "      repository   pmc_date repository_date               species  \\\n",
       "0            GEO  2013-4-18      2000-09-28          Homo sapiens   \n",
       "1            GEO  2013-5-14      2004-05-25          Mus musculus   \n",
       "2            GEO  2014-11-2      2003-11-07          Homo sapiens   \n",
       "3            GEO  2016-1-28      2013-05-06     Rattus norvegicus   \n",
       "4            GEO  2016-1-28      2002-03-11          Homo sapiens   \n",
       "...          ...        ...             ...                   ...   \n",
       "73485        GEO   2019-6-4      2002-03-11          Homo sapiens   \n",
       "73486        GEO   2019-6-2             NaN                   NaN   \n",
       "73487        GEO  2013-11-5      2009-03-16          Homo sapiens   \n",
       "73488        SRA  2019-3-31      2018-03-13  human gut metagenome   \n",
       "73489        SRA  2019-3-31      2018-03-13  human gut metagenome   \n",
       "\n",
       "                      hardware  library_strategy  sraAvg_length    sraBases  \\\n",
       "0                  SAGE NlaIII  Expression_Array            NaN         NaN   \n",
       "1      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "2      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "3      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "4      in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "...                        ...               ...            ...         ...   \n",
       "73485  in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "73486                      NaN               NaN            NaN         NaN   \n",
       "73487  in situ oligonucleotide  Expression_Array            NaN         NaN   \n",
       "73488           Illumina MiSeq          AMPLICON          602.0  17743348.0   \n",
       "73489           Illumina MiSeq          AMPLICON          602.0  17743348.0   \n",
       "\n",
       "      sraAccess  total_use_count reuse_role  \n",
       "0           NaN               64          R  \n",
       "1           NaN                1          G  \n",
       "2           NaN               43          R  \n",
       "3           NaN                1          G  \n",
       "4           NaN               32          R  \n",
       "...         ...              ...        ...  \n",
       "73485       NaN                1          G  \n",
       "73486       NaN               22          R  \n",
       "73487       NaN                5          G  \n",
       "73488    public                1          G  \n",
       "73489    public                1          G  \n",
       "\n",
       "[73490 rows x 15 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge these back onto the original table, matching dates back to papers\n",
    "allFactors = pd.merge(allFactors, justPap, how = 'left', \n",
    "                      left_on = ['converted_accession', 'pmc_date'], \n",
    "                      right_on = ['converted_accession', 'min_pmc_date'])\n",
    "\n",
    "# label every usage of a dataset that isn't 'G' (generator) as 'R' (reuser)\n",
    "allFactors['allR'] = 'R'\n",
    "allFactors['reuse_role'] = allFactors['reuse_role'].fillna(allFactors['allR'])\n",
    "\n",
    "# drop redundant columns\n",
    "allFactors = allFactors.drop(labels = ['min_pmc_date', 'allR'], axis = 1)\n",
    "\n",
    "allFactors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save to .csv\n",
    "allFactors.to_csv('../data_tables/metadataMatrix.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # define function returning corresponding metadata or Study/Series ID given any information on a dataset\n",
    "# def grabRelated(table, key, key_col, out_col):\n",
    "#     indices = table.index[table[key_col] == key].tolist()\n",
    "#     if(len(indices) > 0):\n",
    "#         allHits = table.loc[indices][[out_col]]\n",
    "#         modeHits = allHits.mode()\n",
    "#         if(modeHits.empty):\n",
    "#             return np.NaN\n",
    "#         else:\n",
    "#             return modeHits.iloc[0,0]\n",
    "#     else:\n",
    "#         return np.NaN\n",
    "\n",
    "# # define functions to perform QC on accessions, checking if they exist in the references\n",
    "# def sraChecksOut(acc, col):\n",
    "#     if(acc in sraReference[col].tolist()):\n",
    "#         return True\n",
    "#     else:\n",
    "#         return False\n",
    "    \n",
    "# def geoChecksOut(acc, col):\n",
    "#     if(acc in geoReference[col].tolist()):\n",
    "#         return True\n",
    "#     else:\n",
    "#         return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # check every accession against the corresponding reference, and convert to STUDY/SERIES\n",
    "# # add converted accessions to a new list, to be added as a column later\n",
    "# # CAUTION: long input list, loop takes several hours to run\n",
    "\n",
    "# converted_acc = []\n",
    "\n",
    "# for acc in pmcData['accession']:\n",
    "#     style = ''\n",
    "#     # assign a style to the accession, corresponds to column names\n",
    "#     # SRA styles\n",
    "#     if('SRP' in acc or 'ERP' in acc or 'DRP' in acc):\n",
    "#         style = 'SRAStudy'\n",
    "#     elif('SRR' in acc or 'ERR' in acc or 'DRR' in acc):\n",
    "#         style = 'Run'\n",
    "#     elif('SRX' in acc or 'ERX' in acc or 'DRX' in acc):\n",
    "#         style = 'Experiment'\n",
    "#     elif('PRJNA' in acc or 'PRJD' in acc or 'PRJEB' in acc):\n",
    "#         style = 'BioProject'\n",
    "#     elif('SRA' in acc or 'ERA' in acc or 'DRA' in acc):\n",
    "#         style = 'Submission'\n",
    "#     elif('SRS' in acc or 'ERS' in acc or 'DRS' in acc):\n",
    "#         style = 'Sample'\n",
    "#     # GEO styles\n",
    "#     elif('GSE' in acc):\n",
    "#         style = 'Series'\n",
    "#     elif('GSM' in acc):\n",
    "#         style = 'Accession'\n",
    "#     elif('GPL' in acc):\n",
    "#         style = 'Platform'\n",
    "#     elif('GDS' in acc):\n",
    "#         style = 'Datasets'\n",
    "#     else:\n",
    "#         style = ''\n",
    "        \n",
    "#     # for SRA accessions, check if they exist in the SRA reference\n",
    "#     # ...and add corresponding STUDY ID to new column\n",
    "#     if(style == 'SRAStudy' or style == 'Run' or style == 'Experiment' or \n",
    "#        style == 'BioProject' or style == 'Submission' or style == 'Sample'):\n",
    "#         if(not sraChecksOut(acc, style)):\n",
    "#             converted_acc.append(np.NaN)\n",
    "#         else:\n",
    "#             converted_acc.append(grabRelated(sraReference, acc, style, 'SRAStudy'))\n",
    "#     # for GEO accessions, check if they exist in the GEO reference\n",
    "#     # ...and add corresponding SERIES ID to new column\n",
    "#     elif(style == 'Series' or style == 'Accession' or style == 'Platform' or style == 'Datasets'):\n",
    "#         if(not geoChecksOut(acc, style)):\n",
    "#             converted_acc.append(np.NaN)\n",
    "#         else:\n",
    "#             converted_acc.append(grabRelated(geoReference, acc, style, 'Series'))\n",
    "#     # something doesn't belong, record as missing\n",
    "#     else:\n",
    "#         converted_acc.append(np.NaN)\n",
    "\n",
    "# # add the converted accession list as a new column in PubMed data table\n",
    "# # pd.set_option('display.min_rows', 50)\n",
    "# pmcData.loc[:,'converted_accession'] = converted_acc\n",
    "# pmcData = pmcData.drop_duplicates()\n",
    "# pmcData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save 'checkpoint' to CSV file\n",
    "# pmcData.to_csv('postFilterMatrix.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # temporary cell saving time from big loop...\n",
    "\n",
    "# pmcData = pd.read_csv('../data_tables/postFilterMatrix.csv')\n",
    "# pmcData"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
